### PR TITLE
modules/SceAudio: Implement sceAudioOutSetConfig.

### PR DESCRIPTION
**Pull request by Zangetsu38**

### CODE REVIEW

**Comment by bookmist** [comment]
Great, but I think it would be good to add STUB or TODO.

**Comment by Zangetsu38** [comment]
> Great, but I think it would be good to add STUB or TODO.

Not really, no have really way for set new config in one stream exist, need so obligatory close and re create new port/stream

**Comment by Zangetsu38** [comment]
Ils is ready

**Comment by bookmist** [comment]
len, freq, and param  can be a -1, what mean holds the current value

**Comment by Zangetsu38** [comment]
> len, freq, and param can be a -1, what mean holds the current value

Is not possible read current value, so how you want doing this ?

File: vita3k/modules/SceAudio/SceAudio.cpp [hunk 1]
```diff
@@ -201,7 +206,32 @@ EXPORT(int, sceAudioOutSetCompress) {
...
+        return RET_ERROR(SCE_AUDIO_OUT_ERROR_INVALID_PORT);
+
+    const auto set_len = len > 0 ? len : prt->len;
+    const auto set_freq = freq > 0 ? freq : prt->freq;
+    const auto set_mode = mode > 0 ? mode : prt->mode;
+
+    if (CALL_EXPORT(sceAudioOutReleasePort, port) != 0)
+        return RET_ERROR(SCE_AUDIO_OUT_ERROR_INVALID_PORT);
+
+    const auto channels = (mode == SCE_AUDIO_OUT_MODE_MONO) ? 1 : 2;
```
Comment (bookmist) [inline vita3k/modules/SceAudio/SceAudio.cpp:220]: set_mode is ignored here.

Comment (Zangetsu38) [reply vita3k/modules/SceAudio/SceAudio.cpp:220]: Oups lol

File: vita3k/modules/SceAudio/SceAudio.cpp [hunk 2]
```diff
@@ -201,7 +206,35 @@ EXPORT(int, sceAudioOutSetCompress) {
...
 EXPORT(int, sceAudioOutSetConfig, int port, SceSize len, int freq, SceAudioOutMode mode) {
     TRACY_FUNC(sceAudioOutSetConfig, port, len, freq, mode);
-    return UNIMPLEMENTED();
+    AudioOutPortPtr prt = lock_and_find(port, emuenv.audio.out_ports, emuenv.audio.mutex);
+    if (!prt)
+        return RET_ERROR(SCE_AUDIO_OUT_ERROR_INVALID_PORT);
+
+    const auto set_len = len > 0 ? len : prt->len;
+    const auto set_freq = freq > 0 ? freq : prt->freq;
+    const auto set_mode = mode > 0 ? mode : prt->mode;
```
Comment (bookmist) [inline vita3k/modules/SceAudio/SceAudio.cpp:215]: SCE_AUDIO_OUT_MODE_MONO = 0 and it's valid value. also check that mode is valid

**Review by bookmist — APPROVED** [review]

**Pull request merged**